rust/hyperlane-core/src/traits/indexer.rs

```diff
@@ -54,3 +43,10 @@ pub trait MessageIndexer: Indexer<HyperlaneMessage> + 'static {
     /// Return the latest finalized mailbox count and block number
     async fn fetch_count_at_tip(&self) -> ChainResult<(u32, u32)>;
 }
+
+#[async_trait]
+#[auto_impl(&, Box, Arc)]
+pub trait NonceIndexer<T>: Indexer<T> + 'static {
```

Comment:
nit: for consistency probably best to keep with "sequence" instead of nonce in situations where it doesn't necessarily only relate to message nonces